### PR TITLE
Finish adding examples in docstrings

### PR DESCRIPTION
**Pull request by martaiborra**

### CODE REVIEW

File: caterva2/api.py [hunk 1]
```diff
@@ -1092,10 +1152,19 @@ def remove(self):
...
+        >>> urlbase = 'https://cloud.caterva2.net/demo'
+        >>> root = cat2.Root('@personal', urlbase, dict(username='user@example.com', password='foo'))
+        >>> path = 'root-example/dir2/ds-4d.b2nd'
+        >>> root.upload(path)
+        <Dataset: @personal/root-example/dir2/ds-4d.b2nd>
+        >>> auth_cookie = cat2.get_auth_cookie(urlbase, dict(username='user@example.com', password='foo'))
+        >>> file = cat2.File(path, '@personal', urlbase, auth_cookie)
         >>> file.remove()
-        '@public/ds-1d.b2nd'
+        '@personal/root-example/dir2/ds-4d.b2nd'
```
Comment (FrancescAlted) [inline caterva2/api.py:1170]: For what is worth, here it is an example of using a complete path for identifying a file.  I think this is the way to go.  I'll open a ticket and hopefully we will put some consistency in reported paths through the API eventually.

File: caterva2/api.py [hunk 2]
```diff
@@ -1190,6 +1258,11 @@ def c2context(
     ...     print(cat2.get_roots()['h5lung_j2k'])
     ...
     {'name': 'h5lung_j2k', 'http': 'localhost:8012', 'subscribed': None}
+    >>> urlbase = 'https://cloud.caterva2.net/demo'
+    >>> with cat2.c2context(urlbase=urlbase, username='user@example.com', password='foo'):
+    ...     print(cat2.upload('root-example/ds-2d-fields.b2nd', '@personal/fields.b2nd'))
+    ...
+    @personal/fields.b2nd
```
Comment (FrancescAlted) [inline caterva2/api.py:1270]: Ditto

File: caterva2/api.py [hunk 3]
```diff
@@ -1020,10 +1064,8 @@ def download(self, localpath=None):
         >>> file = root['ds-1d.b2nd']
         >>> file.download()
         PosixPath('example/ds-1d.b2nd')
-        >>> file.download('mypath')  # assuming 'mypath' exists
-        PosixPath('mypath/ds-1d.b2nd')
-        >>> file.download('mypath/myds.b2nd')  # 'mypath' will be created if it doesn't exist
-        PosixPath('mypath/myds.b2nd')
+        >>> file.download('mypath')
+        PosixPath('mypath')
```
Comment (FrancescAlted) [inline caterva2/api.py]: Perhaps a small explanation (a comment would be fine) on that mypath is meant to be a dir.  Better yet, why not using `mydir` instead?

File: caterva2/api.py [hunk 4]
```diff
@@ -654,12 +669,35 @@ def listusers(username=None, urlbase=None, auth_cookie=None):
...
     list of dict
         The list of users in the subscriber.
+
+    Examples
+    --------
+    >>> import caterva2 as cat2
+    >>> import numpy as np
+    >>> # To list the users you need to be a superuser
+    >>> # This example is intended to work when the subscriber is running locally
+    >>> super_auth_cookie = cat2.get_auth_cookie(cat2.sub_urlbase_default, user_auth=dict(username='superuser@example.com', password='foo'))
```
Comment (FrancescAlted) [inline caterva2/api.py]: Line is too long for a pleasant reading.  Can you make it shorter?

File: caterva2/api.py [hunk 5]
```diff
@@ -628,12 +629,26 @@ def deluser(user, urlbase=None, auth_cookie=None):
...
     str
         An explanatory message.
+
+    Examples
+    --------
+    >>> import caterva2 as cat2
+    >>> import numpy as np
+    >>> # To delete a user you need to be a superuser
+    >>> # This example is intended to work when the subscriber is running locally
+    >>> super_auth_cookie = cat2.get_auth_cookie(cat2.sub_urlbase_default, user_auth=dict(username='superuser@example.com', password='foo'))
```
Comment (FrancescAlted) [inline caterva2/api.py]: Line is too long

**Review by FrancescAlted — APPROVED** [review]
Looks good!  I have a few notes, but otherwise, feel free to merge when you are happy.

File: caterva2/api.py (hunk 1 above)
Comment (FrancescAlted) [reply caterva2/api.py:1170]: En the end, only the `api/list` has the behavior of not listing the complete path, but this is on purpose, and only when a parameter is passed.  So, do not pay attention to the above comment.

File: caterva2/api.py (hunk 3 above)
Comment (martaiborra) [reply caterva2/api.py]: Fixed in 0fe79ea0f27976e7d428308106ade05bc649e08b

File: caterva2/api.py (hunk 4 above)
Comment (martaiborra) [reply caterva2/api.py]: Fixed in 0fe79ea0f27976e7d428308106ade05bc649e08b

File: caterva2/api.py (hunk 5 above)
Comment (martaiborra) [reply caterva2/api.py]: Fixed in 0fe79ea0f27976e7d428308106ade05bc649e08b

**Pull request merged**